utils/cross-build-mlir-aie.sh

```diff
@@ -47,31 +43,21 @@ mkdir -p $INSTALL_DIR
 cd $BUILD_DIR
 
 set -o pipefail
-
+set -e
 cmake -GNinja \
-    -DBUILD_SHARED_LIBS=OFF \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} \
-    -DCMAKE_MODULE_PATH=${CMAKEMODULES_DIR} \
-    -DCMAKE_SYSROOT=${CMAKE_SYSROOT} \
-    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} \
+    -DCMAKE_MODULE_PATH=${CMAKEMODULES_DIR}/modulesXilinx \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKEMODULES_DIR}/toolchainFiles/toolchain_clang_crosscomp_pynq.cmake \
+    -DSysroot=${CMAKE_SYSROOT} \
+    -DArch=arm64 \
     -DLLVM_DIR=${LLVM_DIR}/build-aarch64/lib/cmake/llvm \
     -DMLIR_DIR=${LLVM_DIR}/build-aarch64/lib/cmake/mlir \
-    -DVitisSysroot=${CMAKE_SYSROOT} \
+    -DLLVM_USE_LINKER=lld \
+    -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} \
+    -DAIE_ENABLE_BINDINGS_PYTHON=ON \
+    -DCMAKE_BUILD_TYPE=Debug \
+    -Wno-dev \
     .. |& tee cmake.log
 
-ec=$?
-if [ $ec -ne 0 ]; then
-    echo "CMake Error"
-    exit $ec
-fi
-
 ninja install |& tee ninja-install.log
 
-ec=$?
-if [ $ec -ne 0 ]; then
-    echo "Build Error"
-    exit $ec
-fi
-
```

Comment:
Revert the removal of error checking. This is required when I run this script in docker, otherwise it keeps going and I don't know where it failed.

Reply:
If it was useful someone would have commented the use case, right? ;-)
So we are free to remove it? :-)
So, please add a lot of comments stating the hypothesis and whatever for the newcomers for  a more inclusive world.

Reply:
Checking return codes is common good practice. It should not need a comment `// check for error`